ecs_fargate/ecs.tf

```diff
@@ -0,0 +1,237 @@
+  scalable_dimension = aws_appautoscaling_target.main[count.index].scalable_dimension
+  service_namespace  = aws_appautoscaling_target.main[count.index].service_namespace
+
+  target_tracking_scaling_policy_configuration {
+    predefined_metric_specification {
+      predefined_metric_type = "ECSServiceAverageMemoryUtilization"
+    }
+    target_value = var.ecs_auto_scale_memory
+  }
+}
+
+// optional: scale up when cpu is over X %
+resource "aws_appautoscaling_policy" "cpu" {
+  count = length(var.ecs_auto_scale_arn) == 0 ? 0 : var.ecs_auto_scale_cpu == 0 ? 0 : 1
+
+  name               = "${data.aws_ecs_cluster.main.cluster_name}-${aws_ecs_service.main.name}-cpu-autoScalingPolicy"
+  policy_type        = "TargetTrackingScaling"
+  resource_id        = aws_appautoscaling_target.main[count.index].id
+  scalable_dimension = aws_appautoscaling_target.main[count.index].scalable_dimension
+  service_namespace  = aws_appautoscaling_target.main[count.index].service_namespace
+
+  target_tracking_scaling_policy_configuration {
+    predefined_metric_specification {
+      predefined_metric_type = "ECSServiceAverageCPUUtilization"
+    }
+
+    target_value = var.ecs_auto_scale_cpu
+  }
+}
+
+// optional: spin service down on a schedule
+resource "aws_appautoscaling_scheduled_action" "schedule_down" {
+  count              = length(var.ecs_auto_scale_arn) == 0 ? 0 : var.ecs_auto_scale_schedule_down == "0" ? 0 : 1
+  name               = "${data.aws_ecs_cluster.main.cluster_name}-${aws_ecs_service.main.name}-schedule-down"
+  resource_id        = aws_appautoscaling_target.main[count.index].id
+  scalable_dimension = aws_appautoscaling_target.main[count.index].scalable_dimension
+  service_namespace  = aws_appautoscaling_target.main[count.index].service_namespace
+  schedule           = "cron(${var.ecs_auto_scale_schedule_down})"
+
+  scalable_target_action {
+    min_capacity = 0
+    max_capacity = 0
+  }
+}
+
+// optional: spin service up on a schedule
+resource "aws_appautoscaling_scheduled_action" "schedule_up" {
+  count              = length(var.ecs_auto_scale_arn) == 0 ? 0 : var.ecs_auto_scale_schedule_up == "0" ? 0 : 1
+  name               = "${data.aws_ecs_cluster.main.cluster_name}-${aws_ecs_service.main.name}-schedule-down"
+  resource_id        = aws_appautoscaling_target.main[count.index].id
+  scalable_dimension = aws_appautoscaling_target.main[count.index].scalable_dimension
+  service_namespace  = aws_appautoscaling_target.main[count.index].service_namespace
+  schedule           = "cron(${var.ecs_auto_scale_schedule_up})"
+
+  scalable_target_action {
+    min_capacity = var.ecs_min_count
+    max_capacity = var.ecs_max_count
+  }
+}
+
+resource "aws_sns_topic" "cb" {
+  count = var.ecs_circuit_breaker ? 1 : 0
+  name = join("", [var.ecs_service_name, "CB", "Topic"])
+  tags = merge(
+    var.tags,
+    {
+      "Name" = join("", [var.ecs_service_name, "CB", "Topic"])
+    },
+  )
+}
```

Comment:
We tend to like to be able to hook these SNS alerting topics into an sns-to-teams lambda ([teamsalerts module](https://github.com/massgov/mds-terraform-common/tree/1.x/teamsalerts)), so while I think it's perfectly fine that we're creating this topic within the module and attaching 0 or more email subscriptions, I think it'd be helpful to output the topic ARN from the module.